src/gsy_e/models/area/market_rotators.py

```diff
@@ -53,26 +53,31 @@ class ForwardMarketRotatorBase:
     def __init__(self, markets: ForwardMarketBase):
         self.markets = markets
 
+    @staticmethod
+    def _get_last_slot_to_be_deleted(current_time: DateTime) -> DateTime:
+        """
+        Return last time_slot that will be deleted from the ahead market buffer.
+        """
+
     @staticmethod
     def _is_it_time_to_rotate(current_time: DateTime) -> bool:
         """Return True if it is time to rotate markets."""
 
     def rotate(self, current_time: DateTime) -> None:
         """Delete orders in expired day-ahead markets."""
         if self._is_it_time_to_rotate(current_time):
-            slots_deleted = []
-            for delivery_time, market_slot_info in self.markets.open_market_slot_info.items():
-                if market_slot_info.closing_time <= current_time:
-                    self.markets.delete_orders_in_old_future_markets(
-                        last_slot_to_be_deleted=delivery_time)
-                    slots_deleted.append(delivery_time)
-            for slot_deleted in slots_deleted:
-                self.markets.open_market_slot_info.pop(slot_deleted)
+            self.markets.delete_orders_in_old_future_markets(
+                last_slot_to_be_deleted=self._get_last_slot_to_be_deleted(
+                    current_time))
 
 
 class IntradayMarketRotator(ForwardMarketRotatorBase):
     """Handles market rotation for the intraday market block"""
 
+    @staticmethod
+    def _get_last_slot_to_be_deleted(current_time: DateTime) -> DateTime:
+        return current_time + duration(minutes=15)
+
```

Comment:
I thought this was there from the first implementation. Or do I remember incorrectly? Was it removed due to a change in design?

Reply:
Yes, there was an attempt to use the closing time of each market for recycling (essentially to compare the expected closing time of the market with the current time and only close markets when the current time is greater than or equal to the closing time) however due to complications in the unit tests we ditched it again in favor of the old approach.